### PR TITLE
A11Y : Email capture remake and notification banner.

### PR DESCRIPTION
**Pull request by elopezanaya**
## **Thank you for your contribution. Before submitting this PR, please include:**

# DEPENDENCY ON PR https://github.com/microsoft/omnichannel-chat-widget/pull/378
### Id of the task, bug, story, or other reference.
3485957
### Description
_Include a description of the problem to be solved_

A11Y, no success notification after introducing the email address to receive the transcript once the chat ends.

## Solution Proposed
_Detail what is the solution proposed, including links to the design document if required or any other document required to support the solution_

- enabling banner for success case
- enhance texts, to communicate effectively that the email will be sent when the chat ends, not when the user submits the email.
- new function to format template messages

PR related to : https://github.com/microsoft/omnichannel-chat-widget/pull/378

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what test cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

### new dialog 

image

### success notification
image

### Error notification
image

- validated with NVDA and Narrator

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/stories/component-stories/InputValidationPane.stories.tsx [hunk 1]
```diff
@@ -19,15 +19,15 @@ const defaultInputValidationPaneProps: IInputValidationPaneProps = {
         inputValidationPaneAriaLabel: "Email Chat Transcript Pane",
     
         hideTitle: false,
-        titleText: "Email this chat transcript",
+        titleText: "EPlease provide e-mail address to send transcript to.",
```
Comment (kevinlin98z) [inline chat-widget/stories/component-stories/InputValidationPane.stories.tsx]: nit: Please

File: chat-widget/src/components/webchatcontainerstateful/common/defaultProps/defaultMiddlewareLocalizedTexts.ts [hunk 2]
```diff
@@ -28,5 +28,7 @@ export const defaultMiddlewareLocalizedTexts: ILiveChatWidgetLocalizedTexts = {
     MIDDLEWARE_MESSAGE_RETRY: "Retry",
     MIDDLEWARE_BANNER_CHAT_DISCONNECT: "Your conversation has been disconnected. For additional assistance, please start a new chat.",
     THIRD_PARTY_COOKIES_BLOCKED_ALERT_MESSAGE: "Third party cookies are blocked. Reloading this page will start a new conversation.",
-    MIDDLEWARE_BANNER_FILE_IS_MALICIOUS: "{0} has been blocked because the file may contain a malware."
+    MIDDLEWARE_BANNER_FILE_IS_MALICIOUS: "{0} has been blocked because the file may contain a malware.",
+    MIDDLEWARE_BANNER_FILE_EMAIL_ADDRESS_RECORDED_SUCCESS: "Email will be sent after chat ends!",
+    MIDDLEWARE_BANNER_FILE_EMAIL_ADDRESS_RECORDED_ERROR: "Email {0} could not be saved!, try again later."
```
Comment (kevinlin98z) [inline chat-widget/src/components/webchatcontainerstateful/common/defaultProps/defaultMiddlewareLocalizedTexts.ts:33]: nit: punctuation, we have an `!` followed by a `,`

**Review by kevinlin98z — APPROVED** [review]

File: chat-widget/src/common/utils.ts [hunk 3]
```diff
@@ -421,4 +421,20 @@ export const createFileAndDownload = (fileName: string, blobData: string, mimeTy
...
+ * @param values array of values to replace the placeholders
+ * @returns formatted string with replaced values
+ */
+// use of any for values as array of any type is passed
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const  formatTemplateString = (templateMessage : string, values : any) => {
+    return templateMessage.replace(/{(\d+)}/g, (match, index) => {
+        return typeof values[index] !== "undefined" ? values[index] : match;
+    });
 };
```
Comment (akhiloct1510) [inline chat-widget/src/common/utils.ts:440]: Can be more generic and cleaner using below. Please check

function replacePlaceholders(inputString: string, ...values: any[]): string {
  return inputString.replace(/{(\d+)}/g, (match, index) => {
    const valueIndex = parseInt(index, 10);
    if (valueIndex < values.length) {
      return values[valueIndex];
    } else {
      return match; // If the index is out of range, keep the placeholder as is.
    }
  });
}

**Review by nareshme2016 — APPROVED** [review]

File: chat-widget/src/common/utils.ts (hunk 3 above)
Comment (elopezanaya) [reply chat-widget/src/common/utils.ts:440]: Ill stick with my version, from my PoV is cleaner and shorter, all scenarios are included in unit tests

if there is a particular scenario Im not testng please let me know

**Pull request merged**